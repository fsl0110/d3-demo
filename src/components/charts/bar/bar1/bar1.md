### static uses react-faux-dom for enabling snapshots

```jsx
<Bar1 />
```
