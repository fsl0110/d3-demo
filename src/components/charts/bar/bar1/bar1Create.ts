import ReactFauxDOM from "react-faux-dom";
import * as d3 from "d3";

export const createChart = (
  data: any,

  width: number,
  height: number,
  animate?: any
) => {
  const el = ReactFauxDOM.createElement("svg");
  const data2 = [
    {
      y: "red",
      x: 10
    },
    {
      y: "blue",
      x: 20
    },
    {
      y: "green",
      x: 30
    }
  ];

  const x = d3
    .scaleLinear()
    .range([0, 500])
    .domain([0, 40]);

  const xAxisGenerator = d3.axisBottom(x).scale(x);

  const svg = d3
    .select(el)
    .append("g")
    .attr("transform", "translate(50,0)");

  svg
    .append("g")
    .call(xAxisGenerator)
    .attr("transform", "translate(0,100)");

  svg
    .selectAll("rect")
    .data(data)
    .enter()
    .append("rect")
    .attr("x", x(0))
    .attr("y", (d, i) => (i + 1) * 25)
    .attr("width", (d: any, i) => x(d.count / 400))
    .attr("height", "20")
    .style("fill", "aqua");

  return el.toReact();
};
