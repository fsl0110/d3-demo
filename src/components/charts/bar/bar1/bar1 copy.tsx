import React, { PureComponent } from "react";
import ReactFauxDOM from "react-faux-dom";
import { createChart } from "./bar1Create";

export interface Props {
  data: { term: string; count: number }[];
  width: number;
  height: number;
}

export interface State {
  data: { term: string; count: number }[];
  width: number;
  height: number;
}

export class Bar1 extends PureComponent<Props, State> {
  private el: any;
  readonly state: State = {
    data: [],
    width: 600,
    height: 300
  };

  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
    console.log("getDerivedStateFromProps", nextProps);
    if (prevState.data !== nextProps.data) {
      return {
        data: nextProps.data
      };
    }
    return null;
  }

  /*   componentDidMount() {
    window.addEventListener("resize", this.updateDimensions);
    const dimensions: DOMRect = this.el.getBoundingClientRect();
    if (dimensions.width !== this.state.width) {
      this.updateDimensions();
    }
  } */
  /* 
  componentWillUnmount() {
    window.removeEventListener("resize", this.updateDimensions);
  }

  updateDimensions = () => {
    const dimensions: DOMRect = this.el.getBoundingClientRect();
    console.log(dimensions);
    this.setState({
      width: dimensions.width,
      height: dimensions.height
    });
  }; */

  render() {
    const { data, width, height } = this.state;
    console.log("state width:", width);
    console.log("state data:", data);

    this.el = ReactFauxDOM.createElement("svg");
    return <>{createChart(data, width, height)}</>;
  }
}
