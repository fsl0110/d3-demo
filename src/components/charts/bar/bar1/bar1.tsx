import React, { PureComponent } from "react";
import { createChart } from "./bar1Create";

export interface Props {
  data: { term: string; count: number }[];
  width: number;
  height: number;
}

export interface State {
  data: { term: string; count: number }[];
  width: number;
  height: number;
}

export class Bar1 extends PureComponent<Props, State> {
  readonly state: State = {
    data: [],
    width: 350,
    height: 150
  };

  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
    if (prevState.data !== nextProps.data) {
      return {
        data: nextProps.data
      };
    }
    return null;
  }

  render() {
    const { data, width, height } = this.state;
    return <>{createChart(data, width, height)}</>;
  }
}
