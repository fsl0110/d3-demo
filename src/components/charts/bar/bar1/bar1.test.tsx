import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Bar1 } from "./bar1";

it("renders toMatchSnapshot", () => {
  expect(toJson(shallow(<Bar1 />))).toMatchSnapshot();
});
