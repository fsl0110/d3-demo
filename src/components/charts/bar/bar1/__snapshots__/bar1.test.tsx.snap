// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toMatchSnapshot 1`] = `
<Fragment>
  <svg
    key="faux-dom-0"
    style={Object {}}
  >
    <g
      key="faux-dom-0"
      style={Object {}}
      transform="translate(50,0)"
    >
      <g
        fill="none"
        fontFamily="sans-serif"
        fontSize={10}
        key="faux-dom-0"
        style={Object {}}
        textAnchor="middle"
        transform="translate(0,100)"
      >
        <path
          className="domain"
          d="M0.5,6V0.5H500.5V6"
          key="faux-dom-0"
          stroke="currentColor"
          style={Object {}}
        />
        <g
          className="tick"
          key="faux-dom-1"
          opacity={1}
          style={Object {}}
          transform="translate(0.5,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            0
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-2"
          opacity={1}
          style={Object {}}
          transform="translate(63,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            5
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-3"
          opacity={1}
          style={Object {}}
          transform="translate(125.5,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            10
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-4"
          opacity={1}
          style={Object {}}
          transform="translate(188,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            15
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-5"
          opacity={1}
          style={Object {}}
          transform="translate(250.5,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            20
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-6"
          opacity={1}
          style={Object {}}
          transform="translate(313,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            25
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-7"
          opacity={1}
          style={Object {}}
          transform="translate(375.5,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            30
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-8"
          opacity={1}
          style={Object {}}
          transform="translate(438,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            35
          </text>
        </g>
        <g
          className="tick"
          key="faux-dom-9"
          opacity={1}
          style={Object {}}
          transform="translate(500.5,0)"
        >
          <line
            key="faux-dom-0"
            stroke="currentColor"
            style={Object {}}
            y2={6}
          />
          <text
            dy="0.71em"
            fill="currentColor"
            key="faux-dom-1"
            style={Object {}}
            y={9}
          >
            40
          </text>
        </g>
      </g>
      <rect
        height="20"
        key="faux-dom-1"
        style={
          Object {
            "fill": "red",
          }
        }
        width={125}
        x={0}
        y={25}
      />
      <rect
        height="20"
        key="faux-dom-2"
        style={
          Object {
            "fill": "blue",
          }
        }
        width={250}
        x={0}
        y={50}
      />
      <rect
        height="20"
        key="faux-dom-3"
        style={
          Object {
            "fill": "green",
          }
        }
        width={375}
        x={0}
        y={75}
      />
    </g>
  </svg>
</Fragment>
`;
